### PR TITLE
Fix clean command skipping flow cells with one bundle missing

### LINKED ISSUE #2861: Flow cells not removed
## Description

Some flow cell directories are not removed correctly. 

Examples:
22FWKKLT3
22FYY7LT3
22FJMMLT3 - Missing fastq files in housekeepr

## Suggested solution
- Look into the logs (`journalctl --user-unit cg-clean-flow-cells.service`) and see which criteria is blocking the removal
- See if the criterias are too strict

## This can be closed when

Any flow cell, (barring crashed runs) are removed once demultiplexed and added to the databases

### PR DESCRIPTION
**Pull request by diitaz93**
## Description
Fixes #2861. The clean command would fail a check if at least one sample bundle was missing in Housekeeper for the given flow cell.

### Changed

- Changed debug statements for warnings

### Fixed

- The check now fails if **all** samples are missing. So if only one sample is missing, it will clean the existing sample bundles
- A Try-except will catch individual bundles `HousekeeperBundleVersionMissingError`

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b fix-clean-flow-cells -a
    ```

### How to test

- [ ] Copy one of the uncleaned flow cells from prod to stage
- [ ] run `cg clean flow-cells --dry-run`

### Expected test outcome

- [ ] Check that the flow cell would have been deleted

## Review

- [ ] Tests executed by
- [x] "Merge and deploy" approved by CO
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Deploy this branch on 
  - [ ] cg stage
  - [ ] cg production

### CODE REVIEW

File: cg/meta/clean/clean_flow_cells.py [hunk 1]
```diff
@@ -173,9 +170,7 @@ def get_files_for_samples_on_flow_cell_with_tag(self, tag: str) -> list[File] |
                     bundle_name=bundle_name, tags=[tag, self.flow_cell.id]
                 ).all()
             )
-            if not files:
```
Comment (diitaz93) [inline cg/meta/clean/clean_flow_cells.py:176]: Not very clear from here but one level of indentation was removed in this if statement

File: cg/meta/clean/clean_flow_cells.py [hunk 2]
```diff
@@ -161,21 +164,20 @@ def get_sequencing_metrics_for_flow_cell(self) -> list[SampleLaneSequencingMetri
...
-                    f"No files with tag {tag} found for sample {bundle_name} on flow cell {self.flow_cell.id}"
+            try:
+                files.extend(
+                    self.hk_api.get_files_from_latest_version(
+                        bundle_name=bundle_name, tags=[tag, self.flow_cell.id]
+                    ).all()
                 )
-                return None
+            except HousekeeperBundleVersionMissingError:
+                continue
```
[the remark this answers is not in the document]
Comment (ChrOertlin) [reply cg/meta/clean/clean_flow_cells.py:179]: Looks good!

**Review by ChrOertlin — APPROVED** [review]
Nice work and nice refactoring of the tests!

**Comment by diitaz93** [comment]
Renamed branch, solved in #2865

**Pull request closed without merging**